shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -498,8 +528,13 @@ - (BOOL)flt_hasFirstResponderInViewHierarchySubtree {
   // the mask view, whose origin is always (0,0) to the flutter_view.
   CATransform3D reverseTranslate =
       CATransform3DMakeTranslation(-clipView.frame.origin.x, -clipView.frame.origin.y, 0);
-  embedded_view.layer.transform = CATransform3DConcat(finalTransform, reverseTranslate);
-  clipView.maskView = maskView;
+  embedded_view.layer.transform =
+      CATransform3DConcat(GetCATransform3DFromSkMatrix(transformMatrix), reverseTranslate);
+  if (needMask) {
```

Comment:
Optional: You could eliminate `needMask` and just call `clipView.maskView = maskView` each place you currently have `needMask = YES`, so there's less local state to track.